satpy/scene.py

```diff
@@ -158,15 +132,61 @@ def _create_reader_instances(self,
                             reader=reader,
                             reader_kwargs=reader_kwargs)
 
+    @property
+    def sensor_names(self):
+        """Return sensor names for the data currently contained in this Scene.
+
+        If no data is currently loaded or no sensor information is found in
+        the data metadata, loaded readers will be consulted.
```

Comment:
Clarification question: can I have a mixture?  For example, I loaded with sensor X (but have no datasets loaded), and manually add a product from sensor Y (which I didn't load from a file); can I then calculate a composite needing both?  I haven't tested it, but either way, the documentation is somewhat ambiguous here.

Reply:
Good question and good point. First, regarding your review comment about more documentation: I agree. I actually think there might be a part in the sphinx docs that specifically talks about not being able to use load after resampling. I should find this and update this PR.

For your mixture comment, what I expect to happen is that only the "manually" added dataset's sensor would be used. There is no reason that a composite from the manually-added sensor's composite YAML couldn't use a dataset that comes from the readers. However, if the reader's sensor's composite YAML is the one with the composite that depends on both then it wouldn't work.

I'm not sure the best way around this. The Scene either:

1. Never consults the reader
2. Always consults the reader
3. Only consults the reader if no datasets are loaded

I'm not sure if there are any other options that are consistent and not prone to mistakes/bugs. Any other ideas?

Reply:
I'm not sure, either.  I'm happy that I can load from resampled or blended scenes now, and otherwise I'm not sure what's right for mixed or edge cases.  But my comment here was just asking for the documentation to specify the behaviour (so people know what to expect), not for additional functionality (at least not in this PR).

Reply:
I'll see what I can do.

My biggest fear as far as this functionality goes is that a reader might report more sensors than the user is concerned with. If these sensors are included in the composite configs that are loaded then it could result in one of the composites overwriting/overriding the sensor the user is using/focusing on. This is not very likely, but when it does happen it is hard to undo/workaround. Although now that I say that I think we had hoped to be able to load all composite configs in a future redesign of composite loading (easier support of plugins).